language/stdlib/modules/vector.mvir

```diff
@@ -22,8 +22,51 @@ module Vector {
   // Pop an element from the end of vector, abort if the vector is empty.
   native public pop_back<Element>(v: &mut Self.T<Element>): Element;
 
+  // Swaps the elements at the i'th and j'th indices in the vector.
+  native public swap<Element>(v: &mut Self.T<Element>, i: u64, j: u64);
+
+  // Reverses the order of the elements in the vector in place.
+  public reverse<Element>(v: &mut Self.T<Element>) {
+      let front_index: u64;
+      let back_index: u64;
+      let len: u64;
+      front_index = 0;
+      len = Self.length<Element>(freeze(copy(v)));
+
+      if (copy(len) == 0) {
+          return;
+      }
+      assert(copy(len) > 0, 0);
+
+      back_index = move(len) - 1;
+
+      while (copy(front_index) < copy(back_index)) {
+          Self.swap<Element>(copy(v), copy(front_index), copy(back_index));
+          front_index = move(front_index) + 1;
+          back_index = move(back_index) - 1;
+      }
+
+      return;
+  }
+
+  // Moves all of the elements of the `other` vector into the `lhs` vector.
+  // Leaves the `other` vector empty at the end of the call.
+  public append<Element>(lhs: &mut Self.T<Element>, other: &mut Self.T<Element>) {
+      let tmp: Element;
+
+      Self.reverse<Element>(copy(other));
+
+      while (!Self.is_empty<Element>(freeze(copy(other)))) {
+          tmp = Self.pop_back<Element>(copy(other));
+          Self.push_back<Element>(copy(lhs), move(tmp));
```

Comment:
```suggestion
          Self.push_back<Element>(
            copy(lhs),
            Self.pop_back<Element>(copy(other)
          );
```

to eliminate `tmp`?

Reply:
Yes! I need to get used to the more expression-based IR :)